godtools/Views/TractElements/TractBackgroundProperties.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  TractBackgroundProperties.swift
+//  godtools
+//
+//  Created by Pablo Marti on 6/8/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import UIKit
+
+class TractBackgroundProperties: XMLNode {
+    
+    enum BackgroundImageAlign {
+        case center, start, end, top, bottom
+    }
+    
+    enum BackgroundImageScaleType {
+        case fit, fill, fillX, fillY
+    }
+    
+    var backgroundImagePath: String?
+    var backgroundImageAlign: [BackgroundImageAlign]?
+    var backgroundImageScaleType: BackgroundImageScaleType?
+    
+    override func customProperties() -> [String]? {
+        return ["type", "backgroundImageAlign"]
+    }
+    
+    override func performCustomProperty(propertyName: String, value: String) {
+        switch propertyName {
+        case "backgroundImageScaleType":
+            setupBackgroundImageScaleType(value)
+        case "backgroundImageAlign":
+            setupBackgroundImageAlign(value)
+        default: break
+        }
+    }
+    
+    func setupBackgroundImageScaleType(_ string: String) {
+        switch string {
+        case "fit":
+            self.backgroundImageScaleType = .fit
+        case "fill":
+            self.backgroundImageScaleType = .fill
+        case "fill-x":
+            self.backgroundImageScaleType = .fillX
+        case "fill-y":
+            self.backgroundImageScaleType = .fillY
+        default:
+            self.backgroundImageScaleType = .fit
+        }
+    }
+    
+    func setupBackgroundImageAlign(_ string: String) {
```

Comment:
Just in case I change a little this code, basically there are some properties that can not be automatically setup, so there is a format for setting those properties